### ISSUE TITLE
Feature Request : Hierarchical Softmax implementation using Tensorflow

### ISSUE THREAD

**Issue by paduvi** [tensorflow/tensorflow#14749]
I am trying to multi-level classify with Hierarchical Softmax by using Tensorflow. But I could find any existing HS implementation in Tensorflow.

Is there any other way to implement HS using tensorflow?

It would be helpful if Hierarchical Softmax support is given in TF.

**Comment by tatatodd** [comment]
I don't think there is an existing HS implementation in TensorFlow.  Contributions are welcome!

FYI here is the first result from a google search for [[hierarchical softmax tensorflow]](https://www.google.com/search?q=hierarchical+softmax+tensorflow)

https://stackoverflow.com/questions/44140486/scalable-efficient-hierarchical-softmax-in-tensorflow

**Comment by akshaysharma096** [comment]
Hi, I'd like to contribute, some initial guidance for this feature is appreciated. Thanks

**Comment by soloice** [comment]
This is a reference implementation (though it's for balanced complete binary tree): [code](https://github.com/tansey/sdp/tree/87e701c9b0ff3eacab29713cb2c9e7181d5c26aa/tfsdp)

It's released with paper "Deep Nonparametric Estimation of Discrete Conditional Distributions via Smoothed Dyadic Partitioning".

**Comment by dynamicwebpaige** [comment]
Would it make sense to have this added in [TensorFlow Addons](https://www.github.com/tensorflow/addons), instead?

**Comment by soloice** [comment]
HS is not popular nowadays, so maybe there won't be too many people caring about in this.

**Comment by alexander-soare** [comment]
Hi team, I'm a bit new to submitting PRs to mainstream repos. This is tagged as contributions welcome. If I go ahead and build this out and submit a PR will it be looked at? Thanks!
